gsw/tests/test_check_functions.py

```diff
@@ -43,10 +43,12 @@ def cfcf(request):
 def test_check_function(cfcf):
     cv, cf, mfunc = cfcf
     mfunc.run(locals())
-    if mfunc.exception is not None:
+    if mfunc.exception is not None or not mfunc.passed:
         print('\n', mfunc.name)
         print('  ', mfunc.runline)
         print('  ', mfunc.testline)
-        print("   >>>%s<<<" % mfunc.exception)
+        if mfunc.exception is None:
+            mfunc.exception = ValueError('Calculated values are different from the expected matlab results.')
```

Comment:
I think that py.test does provide the values if they come from the failure of an assert.  We need to change the tests to use np.testing.assert_allclose or something like that.  It should be do-able, but I decided to defer it.

I suspect the best way to approach this is to generate the tests more directly from the function names, arguments, and test values from the test data file rather than via translation of the Matlab gsw_check_functions, which is a little bit odd and buggy anyway.  (I'm using a minimally de-bugged version rather than the original.)